hpy/devel/src/runtime/argparse.c

```diff
@@ -27,9 +27,10 @@
  * -------
  *
  * O (object) [HPy *]
- *     Returns an existing handle. Not supported by HPyArg_ParseKeywords as
- *     retrieving an item from the keywords dictionary would require creating
- *     a new handled. Use O+ instead for this case.
+ *     Returns an handle. When using HPyArg_ParseKeywords, this might retrieve
```

Comment:
what about this:
```
Store a handle pointing to a generic Python object. 
To use O, you must provide an HPyTracker. All the handles returned by O are
automatically closed when you call HPyTracker_Close.
```

Reply:
It sounds good, but one only needs to supply an HPyTracker when using `O` with `HPyArg_ParseKeywords`. It's not required when for using `O` with `HPyArg_Parse`. One also only supplies a pointer to an HPyTracker -- not an actual HPyTracker (i.e. it is created for you).

Let me see about improving this bit of documentation along the lines you've suggested though.

Reply:
I wonder whether we should require to pass an `HPyTracker` (which will be unused) also in the `HPyArg_Parse` case, for consistency

Reply:
I thought about that but I'm a little bit -1 on it because it introduces an HPyTracker (and allocating and freeing it) in a case where it's not needed. For the me the current rule is "if an HPyTracker might be required, it will raise an error regardless of the arguments passed if no HPyTracker pointer is given".

Reply:
New documentation added.

This isn't perfect for now -- I intend to add documentation for `HPyArg_Parse` and `HPyArg_ParseKeywords` themselves in a bit (as part of the "documenting the API" project) and hopefully we can polish things more there (it makes more sense to describe how the parameter ht works than to describe the particular case of how `ht` and `O` interact, I think).